File: docs/update_version.py
import re
import os

markdown_file = 'release_notes.md'
version_placeholder = r'{{Version}}'
new_version = os.environ.get('NEW_VERSION')

with open(markdown_file, 'r', encoding='utf8') as file:
    data = file.read()

updated_content = re.sub(version_placeholder, new_version, data)

with open(markdown_file, 'w', encoding='utf8') as file:
    file.write(updated_content)

print('Markdown file updated successfully!')

File: docs/update-version-notes/update_version.js
const fs = require('fs');

const markdownFile = 'release_notes.md';
const versionPlaceholder = /{{ version }}/g;
const newVersion = process.env.NEW_VERSION;

fs.readFile(markdownFile, 'utf8', (err, data) => {
  if (err) {
    console.error(err);
    return;
  }

  const updatedContent = data.replace(versionPlaceholder, newVersion);

  fs.writeFile(markdownFile, updatedContent, 'utf8', (err) => {
    if (err) {
      console.error(err);
      return;
    }

    console.log('Markdown file updated successfully!');
  });
});
